libsnark/gadgetlib1/gadgets/hashes/anemoi/tests/test_anemoi_gadget.cpp

```diff
@@ -0,0 +1,264 @@
+/** @file
+ *****************************************************************************
+ * @author     This file is part of libsnark, developed by Clearmatics Ltd
+ *             (originally developed by SCIPR Lab) and contributors
+ *             (see AUTHORS).
+ * @copyright  MIT license (see LICENSE file)
+ *****************************************************************************/
+
+#include <gtest/gtest.h>
+#include <libff/algebra/curves/bls12_381/bls12_381_init.hpp>
+#include <libff/algebra/curves/bls12_381/bls12_381_pp.hpp>
+#include <libff/common/default_types/ec_pp.hpp>
+#include <libff/common/profiling.hpp>
+#include <libff/common/utils.hpp>
+#include <libsnark/common/default_types/r1cs_ppzksnark_pp.hpp>
+#include <libsnark/gadgetlib1/gadgets/hashes/anemoi/anemoi_components.hpp>
+#include <libsnark/gadgetlib1/gadgets/hashes/anemoi/anemoi_constants.hpp>
+#include <libsnark/zk_proof_systems/ppzksnark/r1cs_gg_ppzksnark/r1cs_gg_ppzksnark.hpp>
+//#include
+//<libsnark/zk_proof_systems/ppzksnark/r1cs_ppzksnark/r1cs_ppzksnark.hpp>
+
+using namespace libsnark;
+
+template<typename ppT> class debug_parameters;
+
+template<> class debug_parameters<libff::bls12_381_pp>
```

Comment:
I don't think we need a template parameter on this class, becauase we know what ppT will be when we use it.  It could also be called `debug_parameters_bls12_381`.

Reply:
Fixed.